### ISSUE TITLE
compute_all_sensitivities(params=None) returns non-independent columns: a primary's column is a total derivative through the derived parameters, whose own columns are in the same tensor, and Result.gradient sums all of them

### ISSUE THREAD

**Issue by wshlavacek** [lanl/bngsim#203]
Split out of #188, which fixed the write semantics but left this. On `eef94d5`.

## The columns are not independent coordinates

`compute_all_sensitivities(params=None)` means "everything computable", and on a model
with derived (expression-backed) parameters that list includes them:

```python
m = bngsim.Model.load("benchmarks/sbml_events/BIOMD0000000701.xml")
r = bngsim.Simulator(m).compute_all_sensitivities((0.0, 100.0), 21, n_workers=1)
# 77 columns, 6 of them derived:
#   _rateLaw_R16_fwd .. _rateLaw_R21_fwd, each = alpha * konB*
```

And a **primary's column is a total derivative through those derived parameters** —
that is exactly the chain rule #188 restored. Proof by construction: pin the six
derived parameters (`set_param(..., force_override=True)`) and `d(x)/d(alpha)` goes
from `1.045e-3` to **exactly 0**, because `alpha` reaches the trajectory only through
them.

So `alpha`'s column already contains the effect that `_rateLaw_R16_fwd`'s column
reports again. `Result.gradient` contracts **all** of them into a single
`(n_params,)` vector, and the docstring hands that to `scipy.optimize.minimize` over a
parameter vector of the same width. The physical effect of `alpha` on R16 is counted
twice.

On this model the derived columns are ~6.7e-12 against alpha's 1.0e-3, so the error is
negligible *here* — the magnitude is model-dependent and nothing bounds it.

## Same hazard the codebase already names elsewhere

#155 established that the parameter axis of an IC sensitivity is a **total** derivative
and must not be summed with the `ic` axis. This is the same shape one level up, between
two columns of the parameter axis itself.

## There is already a pattern for the fix, in the same five lines

The `params is None` branch of `compute_all_sensitivities` (`_simulator.py` ~3753-3778)
*already* filters and warns for exactly this reason — it drops the compartment sizes
`set_param` refuses (#164/#170 stage 3) and names them:

> `compute_all_sensitivities: skipping compartment size(s) [...] — ... Pass params=[...] to make the refusal explicit`

Derived parameters want the same treatment for a different reason: not "this column is
wrong" but "this column is not independent of the others in the tensor". Candidates:

* drop them from the `params=None` default and say so, the way compartment sizes are
  dropped — `Model.primary_param_names` is already exactly the right list, and an
  explicit `params=[...]` keeps the column for anyone who wants `∂x/∂_rateLaw` on its
  own terms;
* or keep them and document the non-independence on `Result.gradient`, which is
  strictly weaker since `gradient` is the function that silently sums them.

The first also makes the default agree with `bngsim.jax.differentiable_solve`, whose
default (`flat=False`) already differentiates over `primary_param_names` and documents
`flat=True` as the opt-in for treating derived parameters as independent axes.

## Related

The original framing I split this from was "there is no warning when `set_param`
overrides a derived parameter". That is the weaker half and probably not worth acting
on by itself: after #188 an override round-trips, and a non-empty `param_expressions`
entry paired with `param_is_expression == False` identifies one exactly. The tensor
default above is where it actually reaches a result a caller will act on.
